App/Views/TrayWindow.xaml.cs

```diff
@@ -257,41 +239,38 @@ private SizeInt32 CalculateWindowSize(double height)
         return new SizeInt32(newWidth, newHeight);
     }
 
-    private PointInt32 CalculateWindowPosition(SizeInt32 size)
+    private PointInt32 CalculateWindowPosition(SizeInt32 panelSize)
     {
-        var width = size.Width;
-        var height = size.Height;
+        var area = DisplayArea.GetFromWindowId(AppWindow.Id, DisplayAreaFallback.Primary);
+        var bounds = area.OuterBounds; // entire monitor rect
+        var workArea = area.WorkArea; // monitor minus taskbar (and other app bars)
 
-        var cursorPosition = _lastActivatePosition;
-        if (cursorPosition is null)
-        {
-            var primaryWorkArea = DisplayArea.Primary.WorkArea;
-            return new PointInt32(
-                primaryWorkArea.Width - width,
-                primaryWorkArea.Height - height
-            );
-        }
-
-        // Spawn the window to the top right of the cursor.
-        var x = cursorPosition.Value.X + 10;
-        var y = cursorPosition.Value.Y - 10 - height;
+        var tb = GetTaskbarInfo(area);
 
-        var workArea = DisplayArea.GetFromPoint(
-            new PointInt32(cursorPosition.Value.X, cursorPosition.Value.Y),
-            DisplayAreaFallback.Primary
-        ).WorkArea;
-
-        // Adjust if the window goes off the right edge of the display.
-        if (x + width > workArea.X + workArea.Width) x = workArea.X + workArea.Width - width;
+        int x, y;
+        switch (tb.Position)
+        {
+            case TaskbarPosition.Left:
+                x = bounds.X + tb.Gap;
+                y = workArea.Y + workArea.Height - panelSize.Height;
+                break;
 
-        // Adjust if the window goes off the bottom edge of the display.
-        if (y + height > workArea.Y + workArea.Height) y = workArea.Y + workArea.Height - height;
+            case TaskbarPosition.Top:
+                x = workArea.X + workArea.Width - panelSize.Width;
+                y = bounds.Y + tb.Gap;
+                break;
 
-        // Adjust if the window goes off the left edge of the display (somehow).
-        if (x < workArea.X) x = workArea.X;
+            case TaskbarPosition.Bottom when tb.AutoHide:
+                // Auto-hide bottom bar sits under the workArea – use workArea, not bounds.
+                x = workArea.X + workArea.Width - panelSize.Width;
+                y = workArea.Y + workArea.Height - panelSize.Height - tb.Gap;
+                break;
 
-        // Adjust if the window goes off the top edge of the display (somehow).
-        if (y < workArea.Y) y = workArea.Y;
+            default: // right or bottom when not auto-hiding
+                x = workArea.X + workArea.Width - panelSize.Width;
+                y = bounds.Y + bounds.Height - panelSize.Height - tb.Gap;
+                break;
```

Comment:
I don't get why we wouldn't just always use `workArea.Height` here, the comment doesn't do a very good job explaining why we wouldn't prefer to always use the workArea.Height.